lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,120 @@
+require 'hammer_cli'
+require 'yaml'
+module HammerCLI
+
+  class DefaultsCommand < HammerCLI::AbstractCommand
+
+    class ListDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'list'
+
+      desc 'List all the defaults parameters'
+      def execute
+        list_all_defaults_message
+        HammerCLI::EX_OK
+      end
+    end
+
+    class AddDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'add'
+
+      desc 'Add a default parameter to config'
+      option "--param-name", "option-name", _("The name of the default option"), :required => true
+      option "--param-val", "option-value", _("The value for the default option")
+      option "--plugin-name", "option-name", _("The plugin name defaults will be generated for")
+
+      def execute
+        begin
+          if option_plugin_name.nil? && option_param_val.nil?
+            bad_input
+          else
+            namespace = ("HammerCLI" + option_plugin_name.split('_').collect!{ |w| w.downcase.capitalize }.join).constantize if option_plugin_name
+            param = option_param_val.nil? ? "from_server_"+namespace.to_s : option_param_val
+            Defaults.add_defaults_to_conf({option_param_name => param})
+            added_default_message({option_param_name.to_s => param.to_s})
+          end
+          rescue NameError => e
+          plugin_prob_message
+          rescue StandardError => e
+          file_not_found_message
+          HammerCLI::EX_CONFIG
+        end
+        HammerCLI::EX_OK
+      end
+    end
+
+    def added_default_message(options = {})
+      options.each do |key, val|
+        print_message("Added -" + key.to_s + " default-option with value -"+ val.to_s)
+      end
+    end
+
+    def plugin_prob_message(options = {})
```

Comment:
Unused options could be removed. BTW did you place the message methods outside the command classes for any reason?